### PR TITLE
flow : eBPF 802.1q/VLAN support

### PR DESCRIPTION
**Pull request by nplanel**

### CODE REVIEW

File: probe/ebpf/flow.c [hunk 1]
```diff
@@ -316,10 +335,16 @@ static void fill_flow(struct __sk_buff *skb, struct flow *flow)
 	fill_link(skb, 0, flow);
 
 	__u16 protocol = load_half(skb, offsetof(struct ethhdr, h_proto));
+	if (protocol == ETH_P_8021Q) {
+		#pragma unroll
+		for(int i=0;i<5;i++) {
+			protocol = full_vlan(skb, ETH_HLEN + (4*i), flow);
```
Comment (lebauce) [inline probe/ebpf/flow.c]: s/full/fill/

File: probe/ebpf/flow.c [hunk 2]
```diff
@@ -28,15 +28,20 @@
...
 // Fowler/Noll/Vo hash
 #define FNV_BASIS ((__u64)14695981039346656037U)
 #define FNV_PRIME ((__u64)1099511628211U)
 
 #define IP_MF     0x2000
 #define IP_OFFSET	0x1FFF
 
+#define MAX_VLAN_LAYERS 5
+struct vlan {
+	__u16		tci;
```
Comment (safchain) [inline probe/ebpf/flow.c]: in "classic" flow we do have the vlanid as part of the network. I think we should report it to have the same behavior

Comment (nplanel) [reply probe/ebpf/flow.c]: in the "classic" way we just report/calculate the ID on the Link layer.

**Comment by nplanel** [comment]
run skydive-functional-tests-backend-orientdb

**Comment by nplanel** [comment]
run skydive-functional-tests-backend-orientdb

**Pull request merged**